collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -257,30 +278,48 @@ public void answerQuestion(FormIndex index, IAnswerData answer) {
         }
     }
 
+    @NonNull
+    @Override
+    public List<SelectChoice> loadSelectChoices(@NonNull FormEntryPrompt prompt) throws FileNotFoundException, XPathSyntaxException, ExternalDataException {
+        return SelectChoiceUtils.loadSelectChoices(prompt, formController);
+    }
+
     @Override
     protected void onCleared() {
         this.answerListener = null;
+        formSessionObserver.cancel();
     }
 
-    private void updateIndex() {
+    public void refresh() {
         currentIndex.setValue(formController.getFormIndex());
     }
 
+    public void exit() {
+        formSessionRepository.clear(sessionId);
```

Comment:
I don't think we **need** to do this, but if we don't we'll end up retaining `FormController` objects which could cause problems for low memory phones.